cub/cub/device/dispatch/kernels/unique_by_key.cuh

```diff
@@ -26,6 +26,32 @@ CUB_NAMESPACE_BEGIN
 
 namespace detail::unique_by_key
 {
+
+struct VSMemHelper
+{
+  template <typename ActivePolicyT, typename... Ts>
+  using VSMemHelperDefaultFallbackPolicyT =
+    vsmem_helper_default_fallback_policy_t<ActivePolicyT, detail::unique_by_key::AgentUniqueByKey, Ts...>;
+
+  template <typename ActivePolicyT, typename... Ts>
+  _CCCL_HOST_DEVICE static constexpr int BlockThreads(ActivePolicyT /*policy*/)
+  {
+    return VSMemHelperDefaultFallbackPolicyT<ActivePolicyT, Ts...>::agent_policy_t::BLOCK_THREADS;
+  }
+
+  template <typename ActivePolicyT, typename... Ts>
+  _CCCL_HOST_DEVICE static constexpr int ItemsPerThread(ActivePolicyT /*policy*/)
+  {
+    return VSMemHelperDefaultFallbackPolicyT<ActivePolicyT, Ts...>::agent_policy_t::ITEMS_PER_THREAD;
+  }
+
+  template <typename ActivePolicyT, typename... Ts>
+  _CCCL_HOST_DEVICE static constexpr ::cuda::std::size_t VSMemPerBlock(ActivePolicyT /*policy*/)
+  {
+    return VSMemHelperDefaultFallbackPolicyT<ActivePolicyT, Ts...>::vsmem_per_block;
+  }
+};
+
```

Comment:
Ideally `VSMemHelper` would be templated on `typename... Ts` to avoid repetition, but for some reason nvcc 12.0 does not like that, so I resorted to making it a template parameter of all the methods, which is redundant and looks a bit uglier, but works.